### ISSUE TITLE
show gradient in percent

### ISSUE THREAD

**Issue by semicarbazid** [nrenner/brouter-web#55]
Dear Norbert,

for me it would be very useful if the elevation display could show the gradient in percent. Maybe a good position would be below the actual height. In addition, it would also be useful if the route statistics would show the maximum gradient.

The point is that I have a limited development with my gear hub and "cost" and "mean cost" don't tell me if there is a huge slope I cannot climb with my equipment!

Yes, I know I can adapt the uphill cost but sometimes it is difficult to find an adequate alternative route so at least I want to be able to judge the calculated route!

What do you think?

Thanks and best wishes,
Bernd

**Comment by michelschinz** [comment]
Yes, some information about the gradient would be very useful, I agree. One could imagine using color to show the gradient, either directly on the elevation graph, or below it.

**Comment by semicarbazid** [comment]
maybe an additional graph like here: http://www.j-berkemeier.de/GPXViewer/ ?

**Comment by dbeniamine** [comment]
I used to use openrunner and the only feature that I miss is indeed to see the percentage in the elevation graph, there is a screenshot of what we con obtain with openrunner which seems pretty nice to me : 

![image](https://user-images.githubusercontent.com/2989133/87542995-30918200-c6a4-11ea-90d4-150c9ab4e431.png)

**Comment by nrenner** [comment]
Current status:
- meanwhile, route color coding has been added (#242) that also colors the route line by incline (bottom left button)
- I have an unfinished local branch that would show incline percentage for current hover position
- in case we switch to [Leaflet.Heightgraph](https://github.com/GIScience/Leaflet.Heightgraph) gradients like above would be included (see also #310)

**Comment by dbeniamine** [comment]
>     * I have an unfinished local branch that would show incline percentage for current hover position
> 
>     * in case we switch to [Leaflet.Heightgraph](https://github.com/GIScience/Leaflet.Heightgraph) gradients like above would be included (see also #310)

Leafleat Heightgraph is exactly what I am looking for but seeing the current hover percentage would already be a big plus.

**Comment by dbeniamine** [comment]
I saw yesterday that there is now a gradient on brouter-web : 

![image](https://user-images.githubusercontent.com/2989133/107802243-5decfe00-6d61-11eb-939d-774589fc5227.png)

So first congratulation an thank you, it's beautiful :clap: :+1: 

With a legend (to see what slope percentage correspond to each color), with that it would be perfect !

**Comment by Atemu** [comment]
The height map gradient colouring is an amazing feature!

Colouring route segments in the map unfortunately isn't too useful for less extreme routes as even 4-6% inclines are merely shown as light yellow while everything less is just always green. It'd be great if the colour scheme could be adjusted to be the same as the height map's.

I'm not sure this is the right place for this but would it be possible to have an overlay where this gradient is applied to every path rather than just the route? I think that could greatly aid us humans find comfortable routes ourselves by eye or do slight adjustments to path-finders' routes. Hill shading exists but it's not sufficient for biking purposes IMO as the levels where hill shading becomes visible are already very high for cycling standards.  
I've found [this](https://ropensci.github.io/slopes/articles/roadnetworkcycling.html) which would be exactly what I want but I'm not sure it's applicable from the technical side.  
An alternative to this could be a hillshade overlay that uses the same colours and grades as the height map gradient.
